tools/perf_util/src/main/java/com/msft/ort_perf_tool/App.java

```diff
@@ -0,0 +1,179 @@
+		// git rev-parse HEAD
+		String commit_id = exec_command(cwd_dir, "git", "rev-parse", "HEAD");
+		String date = exec_command(cwd_dir, "git", "show", "-s", "--format=%ci", commit_id);
+		final SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss Z");
+		java.util.Date commitDate = sdf.parse(date);
+		final SimpleDateFormat simple_date_format = new SimpleDateFormat("yyyy-MM-dd");
+		String batch_id = simple_date_format.format(commitDate);
+		System.out.println(String.format("Commit change date: %s", batch_id));
+
+		// collect all json files list
+		processPerfMetrics(source_dir, perf_metrics, commit_id, batch_id);
+
+		// TODO - add e2e tests later, run it w/ process command
+	}
+
+	private static void processPerfMetrics(Path source_dir, final List<Path> perf_metrics, String commit_id,
+										   String batch_id) throws Exception {
+		try {
+			Connection conn = JdbcUtil.GetConn();
+			// go thru each json file
+			JSONParser jsonParser = new JSONParser();
+			for (Path metrics_json : perf_metrics) {
+				try (FileReader reader = new FileReader(metrics_json.toAbsolutePath().toString())) {
+					// Read JSON file
+					Object obj = jsonParser.parse(reader);
+					loadMetricsIntoMySQL(conn, commit_id, batch_id, (JSONObject) obj);
+				}
+			}
+		} finally {
+			deletePath(source_dir);
+		}
+	}
+
+	private static void deletePath(Path temp_path) throws IOException {
+		Files.walkFileTree(temp_path, new SimpleFileVisitor<Path>() {
+			@Override
+			public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+				String filename = file.getFileName().toString();
+				if (!filename.startsWith(".") && filename.endsWith(".json")) {
+					Files.deleteIfExists(file);
+				}
+
+				Files.deleteIfExists(file);
+				return FileVisitResult.CONTINUE;
+			}
+		});
+	}
+
+	static private void loadMetricsIntoMySQL(java.sql.Connection conn, String commit_id, String batch_id,
+											 JSONObject json_object) throws Exception {
+
+		try (java.sql.PreparedStatement st = conn.prepareStatement(
+				"INSERT INTO perf_test_training_data (BatchId,CommitId,Model,ModelName,DisplayName,UseMixedPrecision,Optimizer,BatchSize,SeqLen,PredictionsPerSeq," +
+						"NumOfBatches,WeightUpdateSteps,Round,GradAccSteps,AvgTimePerBatch,Throughput,TotalTime,AvgCPU,Memory,RunConfig,Time) " +
+						"values (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,Now())"
+						+ "  ON DUPLICATE KEY UPDATE AvgTimePerBatch=?,Throughput=?,TotalTime=?,AvgCPU=?,Memory=?")) {
+
+			// unique key section
+			st.setString(1, batch_id);
+			st.setString(2, commit_id.substring(0, 8));
+			st.setString(3, (String) json_object.get("Model"));
+			st.setString(4, (String) json_object.get("ModelName"));
+			st.setString(5, (String) json_object.get("DisplayName"));
+			st.setBoolean(6, (Boolean) json_object.get("UseMixedPrecision"));
+			st.setString(7, (String) json_object.get("Optimizer"));
+			st.setInt(8, (int)(long) json_object.get("BatchSize"));
```

Comment:
why do we need two casts?

Reply:
MySQL api workaround. json.get is giving us long, but MySQL has only setInt